specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts

```diff
@@ -335,10 +335,18 @@ export class BusinessRuleManager<SCHEMA extends AnySchema> {
             value: otherValue,
           } = otherFieldValues;
           const { id, cid, value } = fieldValues[fieldName];
+          const field = this.resource.specifyTable.getField(fieldName);
           if (otherCid !== undefined && cid !== undefined && otherCid === cid)
             return true;
           if (otherId !== undefined && id !== undefined && otherId === id)
             return true;
+          if (
+            field !== undefined &&
+            !(field.isRequired || field.localization.isrequired) &&
```

Comment:
the `field.localization.isrequired` check is redundant as it's already included in `field.isRequired`

https://github.com/specify/specify7/blob/b331ce9c552e90d86fc4811a7089fab6ac29c07e/specifyweb/frontend/js_src/lib/components/DataModel/specifyField.ts#L134-L139

Reply:
Currently the `field.localization.isrequired` is not present in `field.isRequired`. 
The `fieldDefinition` only contains information from the `specify_datamodel.xml` (https://files.specifysoftware.org/schema/version/2.10/), and does not contain any information on the splocalecontaineritem which represents the disclipline's configuration for the field. 

Consider the two CollectionObject fields `uniqueIdentifier` and `totalValue`. 
In the currently logged in discipline, `totalValue` is marked as required and  `uniqueIdentifier` is not required. 

The following code is added to the constructor of `FieldBase`
```ts
if (
      table.name === 'CollectionObject' &&
      ['totalvalue', 'uniqueidentifier'].includes(
        fieldDefinition.name.toLowerCase()
      )
    )
      console.log(
        table.name,
        fieldDefinition.name,
        table.localization.items[fieldDefinition.name.toLowerCase()]
          ?.isrequired,
        fieldDefinition.required,
        this.isRequired
      );
 ```

which outputs: 

<img width="454" alt="Screenshot 2024-04-01 at 9 22 48 AM" src="https://github.com/specify/specify7/assets/64045831/d2c633f7-fc9e-4325-adc8-fb4c16018d1a">

---

I agree it would make sense to include the `field.localization.isrequired` with the `field.isRequired`.

Reply:
huh, you are correct. then this seems like a bug as the way it stands now, sp7 ignores schema localization completely for the isRequired setting.

could you please fix it? (in this PR or separate)

Reply:
PR #4721